### ISSUE TITLE
alb-ingress-controller seg faults when creating ingress resources

### ISSUE THREAD

**Issue by justatad** [kubernetes-sigs/aws-alb-ingress-controller#365]
Hi there - We are experiencing an issue where the alb-ingress-controller seg faults without creating any AWS ingress resources or Route 53 config:

```
I0416 09:23:54.452667       1 launch.go:112] &{ALB Ingress Controller 1.0-alpha.9 git-245bb343 git://github.com/coreos/alb-ingress-controller}
I0416 09:23:54.452972       1 launch.go:282] Creating API client for https://100.64.0.1:443
I0416 09:23:54.464562       1 launch.go:295] Running in Kubernetes Cluster version v1.8 (v1.8.7) - git (clean) commit b30876a5539f09684ff9fde266fda10b37738c9c - platform linux/amd64
I0416 09:23:54.467234       1 launch.go:134] validated kube-system/default-http-backend as the default backend
I0416 09:23:54.472419       1 alb-controller.go:128] [ALB-INGRESS] [controller] [DEBUG]: Lock was available. Attempting sync
I0416 09:23:54.472452       1 albingresses.go:91] [ALB-INGRESS] [ingresses] [INFO]: Building list of existing ALBs
I0416 09:23:54.473874       1 session.go:35] [ALB-INGRESS] [session] [INFO]: Request: elasticloadbalancing/DescribeLoadBalancers, Payload: {}
I0416 09:23:54.531573       1 albingresses.go:100] [ALB-INGRESS] [ingresses] [INFO]: Fetching information on 0 ALBs
I0416 09:23:54.531596       1 albingresses.go:176] [ALB-INGRESS] [ingresses] [INFO]: Assembled 0 ingresses from existing AWS resources in 59.114794ms
I0416 09:23:54.531681       1 controller.go:1359] starting Ingress controller
I0416 09:23:54.534403       1 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"default", Name:"salutations", UID:"41c6a9e4-414e-11e8-ada8-025f488373b2", APIVersion:"extensions", ResourceVersion:"670059", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress default/salutations
I0416 09:23:55.402210       1 leaderelection.go:174] attempting to acquire leader lease...
I0416 09:23:55.403757       1 controller.go:477] backend reload required
I0416 09:23:55.404229       1 session.go:35] [ALB-INGRESS] [session] [INFO]: Request: ec2/DescribeSecurityGroups, Payload: {  GroupIds: ["sg-d6b659ab"]}
I0416 09:23:55.405123       1 status.go:193] new leader elected: alb-ingress-controller-cfff45d65-kt7jv
I0416 09:23:55.442480       1 session.go:35] [ALB-INGRESS] [session] [INFO]: Request: ec2metadata/GetDynamicData, Payload: <invalid value>
I0416 09:23:55.444323       1 session.go:35] [ALB-INGRESS] [session] [INFO]: Request: ec2/DescribeInstances, Payload: {  InstanceIds: ["i-0fc1303282832bc9c"]}
I0416 09:23:55.476986       1 session.go:35] [ALB-INGRESS] [session] [INFO]: Request: ec2/DescribeSubnets, Payload: {  SubnetIds: ["subnet-2ae24862","subnet-4c50b416","subnet-719b3e17"]}
E0416 09:23:55.533541       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/asm_amd64.s:514
/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/panic.go:489
/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/panic.go:63
/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/signal_unix.go:290
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/alb/targetgroups/targetgroups.go:143
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/albingress/albingress.go:166
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/albingresses/albingresses.go:72
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/controller/alb-controller.go:162
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/controller/alb-controller.go:141
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller/controller.go:479
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller/controller.go:175
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task/queue.go:89
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task/queue.go:49
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task/queue.go:49
/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/asm_amd64.s:2197
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x15ed4b8]

goroutine 107 [running]:
github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x1806cc0, 0x26deca0)
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/panic.go:489 +0x2cf
github.com/coreos/alb-ingress-controller/pkg/alb/targetgroups.NewDesiredTargetGroups(0xc4209b58b0, 0xc4203027d0, 0x1, 0x1, 0x2, 0xc4205ddfe0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/alb/targetgroups/targetgroups.go:143 +0x78
github.com/coreos/alb-ingress-controller/pkg/albingress.NewALBIngressFromIngress(0xc42031e4e0, 0x0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/albingress/albingress.go:166 +0x9a9
github.com/coreos/alb-ingress-controller/pkg/albingresses.NewALBIngressesFromIngresses(0xc4209b5ac8, 0xc4202920f0, 0x1, 0x1)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/albingresses/albingresses.go:72 +0x285
github.com/coreos/alb-ingress-controller/pkg/controller.(*ALBController).update(0xc4202920f0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/controller/alb-controller.go:162 +0x2c1
github.com/coreos/alb-ingress-controller/pkg/controller.(*ALBController).OnUpdate(0xc4202920f0, 0xc42062fd50, 0x2, 0x2, 0xc42062fd60, 0x2, 0x2, 0x2713038, 0x0, 0x0, ...)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/controller/alb-controller.go:141 +0x2b
github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller.(*GenericController).syncIngress(0xc420522a00, 0x17200e0, 0xc4206dcc80, 0xc4206dcc80, 0x0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller/controller.go:479 +0x867
github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller.(*GenericController).(github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller.syncIngress)-fm(0x17200e0, 0xc4206dcc80, 0xa, 0xc4202825f8)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller/controller.go:175 +0x3e
github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task.(*Queue).worker(0xc420368210)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task/queue.go:89 +0xef
github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task.(*Queue).(github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task.worker)-fm()
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task/queue.go:49 +0x2a
github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4202827a8)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4209b5fa8, 0x3b9aca00, 0x0, 0x127c901, 0xc4204fba40)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4202827a8, 0x3b9aca00, 0xc4204fba40)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task.(*Queue).Run(0xc420368210, 0x3b9aca00, 0xc4204fba40)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/task/queue.go:49 +0x5e
created by github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller.GenericController.Start
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/vendor/k8s.io/ingress/core/pkg/ingress/controller/controller.go:1390 +0x4f5
```

The ingress definition for our application is as follows:

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: salutations
  annotations:
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/subnets: subnet-4c50b416, subnet-2ae24862, subnet-719b3e17
    alb.ingress.kubernetes.io/security-groups: sg-d6b659ab
    #alb.ingress.kubernetes.io/subnets: subnet-1b42a641, subnet-1b42a641, subnet-d1983db7
spec:
  rules:
  - host: salutations.zigzag-london.net
  - http:
      paths:
      - path: /
        backend:
          serviceName: salutations
          servicePort: 80
```

Please let me know if you need any further information from me so we can work through the cause of this issue.

Thanks,

Matthew

**Comment by bigkraig** [comment]
This should be fixed in #403

**Comment by bigkraig** [comment]
Please test with `1.0-alpha.9` and let me know how it goes
